azkaban-common/src/main/resources/azkaban/container/models/v1Service.yaml

```diff
@@ -0,0 +1,22 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: fc-svc-${execid}
```

Comment:
By providing the replacement values as strings we will miss out on any type-checking that could be performed as by a builder.
Even if we use a map for the string substitution could we overlay with it our own builder, something like `AzKubernetesV1ServiceBuilder` and provide methods like
```
builder.withExeciId(execid)
.withPort(port)
.withNamespace(namespace)
...
.build()
``` 
This will provide type safety as well as keep the interface in conformance with what's done for the PodSpec.

Reply:
+1

Reply:
Thanks for the suggestion. Updated the PR with the same.